### DISCUSSION TITLE
Runtime error coming from @mui/materia

### DISCUSSION THREAD

**Discussion by brillout** [vikejs/vike#700] [Discussion]
Moved from Discord:

> Hi, I'm using vite-ssr-plugin in react-ts mode and added material ui, 
single component will render and work fine but I tried to add <ThemeWrapper > to PageShell and then I started to get error when importing files from @mui/materia at runtime

**Comment by brillout** [comment]
What's the error you get?

**Reply by asheghi** [reply]
if I import anything like this 
`
import {createTheme} from "@mui/material/styles/createTheme";
`
I get this error:
`
Error: ERR_UNSUPPORTED_DIR_IMPORT ...\vite-ssr-app\node_modules\@mui\material\styles from ...\vite-ssr-app\dist\server\renderer\default-page-server.js
`

and I import like this
`import {createTheme} from "@mui/material/styles/createTheme.js";`
I get this:
`RollupError: "createTheme" is not exported `

**Reply by brillout** [reply]
Seems like `@mui` is wrongly published. Check out the new docs about this: https://vite-plugin-ssr.com/common-issues#libraries-containing-invalid-esm. Let me know if the solution resolves your issue.

**Reply by asheghi** [reply]
adding @mui/material and @emotion/styled" to noExternal fixed the problem, Thanks!
```
const config: UserConfig = {
  plugins: [react(), ssr()],
  ssr: {
    noExternal: ["@mui/material", "@emotion/styled"],
  },
};
```
